runbot_travis2docker/models/runbot_build.py

```diff
@@ -0,0 +1,283 @@
+    docker_image = fields.Char()
+    docker_container = fields.Char()
+    uses_weblate = fields.Boolean(help='Synchronize with weblate', copy=False)
+    docker_executed_commands = fields.Boolean(
+        help='True: Executed "docker exec CONTAINER_BUILD custom_commands"',
+        readonly=True, copy=False)
+
+    def _get_docker_image(self):
+        self.ensure_one()
+        git_obj = GitRun(self.repo_id.name, '')
+        image_name = git_obj.owner + '-' + git_obj.repo + ':' + \
+            self.name[:7] + '_' + os.path.basename(self.dockerfile_path)
+        return image_name.lower()
+
+    def _get_docker_container(self):
+        self.ensure_one()
+        return "build_%d" % (self.sequence)
+
+    def _job_10_test_base(self, build, lock_path, log_path):
+        if not build.branch_id.repo_id.is_travis2docker_build:
+            return super(RunbotBuild, self)._job_10_test_base(
+                build, lock_path, log_path)
+        build._log('test_base', 'Start test base module runbot_travis2docker')
+        if not build.docker_image or not build.dockerfile_path \
+                or build.result == 'skipped':
+            _logger.info('docker build skipping job_10_test_base')
+            return MAGIC_PID_RUN_NEXT_JOB
+        cmd = [
+            'docker', 'build',
+            "--no-cache", '--pull',
+            "-t", build.docker_image,
+            build.dockerfile_path,
+        ]
+        return self._spawn(cmd, lock_path, log_path, cpu_limit=1200)
+
+    def _job_20_test_all(self, build, lock_path, log_path):
+        if not build.branch_id.repo_id.is_travis2docker_build:
+            return super(RunbotBuild, self)._job_20_test_all(
+                build, lock_path, log_path)
+        if not build.docker_image or not build.dockerfile_path \
+                or build.result == 'skipped':
+            _logger.info('docker build skipping job_20_test_all')
+            return MAGIC_PID_RUN_NEXT_JOB
+        subprocess.call(['docker', 'rm', '-vf', build.docker_container])
+        return self._spawn(
+            build._get_run_cmd(), lock_path, log_path, cpu_limit=1200,
+        )
+
+    def _job_21_coverage(self, build, lock_path, log_path):
+        if not build.branch_id.repo_id.is_travis2docker_build:
+            return super(RunbotBuild, self)._job_21_coverage(
+                build, lock_path, log_path)
+        _logger.info('docker build skipping job_21_coverage')
+        return MAGIC_PID_RUN_NEXT_JOB
+
+    def _job_30_run(self, build, lock_path, log_path):
+        if not build.branch_id.repo_id.is_travis2docker_build:
+            return super(RunbotBuild, self)._job_30_run(
+                build, lock_path, log_path)
+        if (not build.docker_image or not build.dockerfile_path or
+                build.result == 'skipped'):
+            _logger.info('docker build skipping job_30_run')
+            return MAGIC_PID_RUN_NEXT_JOB
+
+        # Start copy and paste from original method (fix flake8)
```

Comment:
why don't you call super here and patch `_spawn` such that is runs another command when it's a docker build and the command is what core would do here?

Reply:
This is a good idea but the main parameter `build` of this job is not received for the method `spawn`
This could be possible if odoo guys use `self` instead of `build` or send this parameter to `spawn` method.